pkg/manifests/manifests.go

```diff
@@ -2844,11 +2845,12 @@ func (f *Factory) ThanosRulerCustomResource(trustedCA *v1.ConfigMap) (*monv1.Tha
 		t.Spec.Tolerations = f.config.ThanosRulerConfig.Tolerations
 	}
 
-	t.Spec.Containers[0].Image = f.config.Images.OauthProxy
+	//images["oauth-proxy"]
+	t.Spec.Containers[1].Image = f.config.Images.OauthProxy
 	setEnv := func(name, value string) {
-		for i := range t.Spec.Containers[0].Env {
-			if t.Spec.Containers[0].Env[i].Name == name {
-				t.Spec.Containers[0].Env[i].Value = value
+		for i := range t.Spec.Containers[1].Env {
+			if t.Spec.Containers[1].Env[i].Name == name {
+				t.Spec.Containers[1].Env[i].Value = value
```

Comment:
i think we added constants somewhere else to identify contained indices

Reply:
I would still prefer to search via name? :) WDYT?

Reply:
oh, yes, sure thing, feel free to implement as a follow-up